# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmTypes.h
# Commit: aeacd48f565e
# Full Hash: aeacd48f565e73a85fff14ef1b109c5964437521
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-11-03 21:39:33
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 3: Implement the algorithm for obtaining tls and use it for wasm signal handling. r=lth
#   
#   This is the third part of series of patches to Frame without tls pointer.
#   Here we preserve initial tls in all entry stubs and then use it to find a proper tls instance for a given frame.
#   
# ==============================================================================

diff -r a5cd8e8ce4b5 -r aeacd48f565e js/src/wasm/WasmTypes.h
--- a/js/src/wasm/WasmTypes.h	Tue Nov 03 10:20:49 2020 +0000
+++ b/js/src/wasm/WasmTypes.h	Tue Nov 03 10:22:41 2020 +0000
@@ -2585,6 +2585,7 @@
   }
   uint32_t lineOrBytecode() const { return lineOrBytecode_; }
   Kind kind() const { return Kind(kind_); }
+  bool mightBeCrossInstance() const { return kind() == CallSiteDesc::Dynamic; }
 };
 
 class CallSite : public CallSiteDesc {
@@ -3304,10 +3305,13 @@
 static_assert(!std::is_polymorphic_v<Frame>, "Frame doesn't need a vtable.");
 
 class FrameWithTls : public Frame {
- public:
   TlsData* calleeTls_;
   TlsData* callerTls_;
 
+ public:
+  TlsData* calleeTls() { return calleeTls_; }
+  TlsData* callerTls() { return callerTls_; }
+
   constexpr static uint32_t sizeWithoutFrame() {
     return sizeof(wasm::FrameWithTls) - sizeof(wasm::Frame);
   }
